api/clients/v2/payloadretrieval/relay_payload_retriever.go

```diff
@@ -65,25 +66,57 @@ func NewRelayPayloadRetriever(
 // verified prior to calling this method.
 func (pr *RelayPayloadRetriever) GetPayload(
 	ctx context.Context,
-	eigenDACert *coretypes.EigenDACertV3) (*coretypes.Payload, error) {
+	eigenDACert coretypes.EigenDACert) (*coretypes.Payload, error) {
+
+	var blobKey *core.BlobKey
+	var relayKeys []core.RelayKey
+	var blobCommitments *encoding.BlobCommitments
+	var err error
+
+	switch eigenDACert.Version() {
+	case coretypes.VersionTwoCert:
+		eigenDAV2Cert, ok := eigenDACert.(*coretypes.EigenDACertV2)
+		if !ok {
+			return nil, fmt.Errorf("expected eigenDACert to be of type *coretypes.EigenDAV2Cert, got %T", eigenDACert)
```

Comment:
I think we can switch directly on the type? Which would remove the need for this `!ok` case (which really should never happen...). We'd get better type safety and cleaner code. Haven't tried to do it though

Reply:
[97e1641](https://github.com/Layr-Labs/eigenda/pull/1607/commits/97e1641e24de5dbfeb50f7ff41cb6e23529f4ab9)